File: src-tauri/src/models/network_config.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[derive(Default)]
pub struct NetworkConfigRow {
    pub blueprint: Option<String>,
    pub server_label: Option<String>,
    pub is_external: Option<bool>,
    pub server_tags: Option<String>,
    pub link_group_ifname: Option<String>,
    pub link_group_lag_mode: Option<String>,
    pub link_group_ct_names: Option<String>,
    pub link_group_tags: Option<String>,
    pub link_speed: Option<String>,
    pub server_ifname: Option<String>,
    pub switch_label: Option<String>,
    pub switch_ifname: Option<String>,
    pub link_tags: Option<String>,
    pub comment: Option<String>,
}

File: src/types/networkConfig.ts
export interface NetworkConfigRow {
  blueprint?: string;
  server_label?: string;
  is_external?: boolean;
  server_tags?: string;
  switch_tags?: string;
  link_group_ifname?: string;
  link_group_lag_mode?: string;
  link_group_ct_names?: string;
  link_group_tags?: string;
  link_speed?: string;
  server_ifname?: string;
  switch_label?: string;
  switch_ifname?: string;
  link_tags?: string;
  comment?: string;
}

export interface ValidationError {
  row_index: number;
  field: string;
  message: string;
}

export interface DataValidationResult {
  valid_rows: NetworkConfigRow[];
  invalid_rows: NetworkConfigRow[];
  errors: ValidationError[];
  duplicates_removed: number;
}